src/kernel/include/thd.h

```diff
@@ -153,6 +166,17 @@ thd_rcvcap_sched(struct thread *t)
 }
 
 static void
+thd_next_thdinfo_update(struct cos_cpu_local_info *cli, struct thread *thd, 
+			struct tcap *tc, tcap_prio_t prio)
+{
+	struct next_thdinfo *nti = (struct next_thdinfo *)cli->next_ti;
```

Comment:
Indeed that is worth worrying about.  However, I think that max 24 bytes (on a 64 bit machine) is fine.  We don't use much stack in the kernel.